### PR TITLE
HomeWork

### PR DESCRIPTION
**Pull request by Daniluko**

### CODE REVIEW

File: Lesson10/src/main/java/dogShelter/DogsServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public class DogsServiceImpl implements DogsService{
+private static Map<Integer,DogsCreator> dogsCreator = Arrays.asList(DogsCreator.creatorDogsShelter()).stream().collect(Collectors.toMap(DogsCreator::getId , Function.identity()));
+
+    @Override
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getDogsInShelter(@PathParam("id") int id) {
+        DogsCreator dogShelter = dogsCreator.get(id);
+        if (dogShelter != null){
+            return Response.status(Response.Status.OK).entity(dogsCreator.get(id)).type(MediaType.APPLICATION_JSON_TYPE).build();
```
Comment (Cherkashyn) [inline Lesson10/src/main/java/dogShelter/DogsServiceImpl.java]: Maybe you should return just .entity(dogShelter)?

File: Lesson10/src/main/java/dogShelter/DogsServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    @Override
+    @PUT
```
Comment (Cherkashyn) [inline Lesson10/src/main/java/dogShelter/DogsServiceImpl.java]: I think this method should be POST. You are inserting a value here.

File: Lesson10/src/main/java/dogShelter/DogsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        DogsCreator dogShelter = dogsCreator.get(id);
+        if (dogShelter != null){
+            dogShelter.getDogs().addAll(Arrays.asList(dog));
+            return Response.status(Response.Status.OK).entity(dogShelter).type(MediaType.APPLICATION_JSON_TYPE).build();
+        }
+        return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    @Override
+    @POST
```
Comment (Cherkashyn) [inline Lesson10/src/main/java/dogShelter/DogsServiceImpl.java]: I think it should be PUT - you are updating here.

File: Lesson10/src/main/java/dogShelter/DogsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    @Path("/{name}")
```
Comment (Cherkashyn) [inline Lesson10/src/main/java/dogShelter/DogsServiceImpl.java]: You missed @Produces annotation :)

File: Lesson10/src/main/webapp/WEB-INF/web.xml [hunk 5]
```diff
@@ -19,4 +19,19 @@
         <url-pattern>/rest/*</url-pattern>
     </servlet-mapping>
 
+    <servlet>
+        <servlet-name>dogShelter</servlet-name>
+        <servlet-class>org.glassfish.jersey.servlet.ServletContainer</servlet-class>
+        <init-param>
+            <param-name>jersey.config.server.provider.packages</param-name>
+            <param-value>dogShelter</param-value>
```
Comment (Cherkashyn) [inline Lesson10/src/main/webapp/WEB-INF/web.xml]: Is it name of your package?

Comment (Daniluko) [reply Lesson10/src/main/webapp/WEB-INF/web.xml]: yes

File: Lesson10/src/main/java/dogShelter/DogsServiceImpl.java (hunk 4 above)
Comment (Daniluko) [reply Lesson10/src/main/java/dogShelter/DogsServiceImpl.java]: i don't need this annotation:)

File: Lesson10/src/main/java/dogShelter/DogsServiceImpl.java (hunk 1 above)
Comment (SergeyPasko) [reply Lesson10/src/main/java/dogShelter/DogsServiceImpl.java]: Yes it more logic

File: Lesson10/src/main/java/dogShelter/DogsServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Override
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response addDog(@PathParam("id") int id, Dog dog) {
+        DogsCreator dogShelter = dogsCreator.get(id);
+        if (dogShelter != null){
+            dogShelter.getDogs().addAll(Arrays.asList(dog));
```
Comment (SergeyPasko) [inline Lesson10/src/main/java/dogShelter/DogsServiceImpl.java]: you have only one dog. Illogical construction

**Pull request merged**